net/uacme/files/run.sh

```diff
@@ -0,0 +1,396 @@
+    if [ -e "/usr/sbin/ip6tables" ]; then
+	ip6tables -D input_rule -p tcp --dport 80 -j ACCEPT -m comment --comment "ACME" 2>/dev/null
+    fi
+    if [ -e /etc/init.d/uhttpd ] && [ "$UPDATE_UHTTPD" -eq 1 ]; then
+	uci commit uhttpd
+	/etc/init.d/uhttpd reload
+	log "Restarting uhttpd..."
+    fi
+
+    if [ -e /etc/init.d/nginx ] && ( [ "$NGINX_WEBSERVER" -eq 1 ] || [ "$UPDATE_NGINX" -eq 1 ] ); then
+	NGINX_WEBSERVER=0
+	/etc/init.d/nginx restart
+	log "Restarting nginx..."
+    fi
+
+    if [ -e /etc/init.d/haproxy ] && [ "$UPDATE_HAPROXY" -eq 1 ]; then
+	/etc/init.d/haproxy restart
+	log "Restarting haproxy..."
+    fi
+}
+
+err_out()
+{
+    post_checks
+    exit 1
+}
+
+int_out()
+{
+    post_checks
+    trap - INT
+    kill -INT $$
+}
+
+is_staging()
+{
+    local main_domain="$1"
+
+    grep -q "acme-staging" "$STATE_DIR/$main_domain/${main_domain}.conf"
+    return $?
+}
+
+issue_cert()
+{
+    local section="$1"
+    local acme_args=
+    local debug=
+    local enabled
+    local use_staging
+    local update_uhttpd
+    local update_nginx
+    local update_haproxy
+    local keylength
+    local domains
+    local main_domain
+    local failed_dir
+    local webroot
+    local dns
+    local ret
+    local staging=
+    local HOOK=
+
+    config_get_bool enabled "$section" enabled 0
+    config_get_bool use_staging "$section" use_staging
+    config_get_bool update_uhttpd "$section" update_uhttpd
+    config_get_bool update_nginx "$section" update_nginx
+    config_get_bool update_haproxy "$section" update_haproxy
+    config_get domains "$section" domains
+    config_get keylength "$section" keylength
+    config_get webroot "$section" webroot
+    config_get dns "$section" dns
+
+    UPDATE_NGINX=$update_nginx
+    UPDATE_UHTTPD=$update_uhttpd
+    UPDATE_HAPROXY=$update_haproxy
+
+    [ "$enabled" -eq "1" ] || return
+
+    if [ "$APP" = "uacme" ]; then
+	[ "$DEBUG" -eq "1" ] && debug="--verbose --verbose"
+    elif [ "$APP" = "acme" ]; then
+	[ "$DEBUG" -eq "1" ] && acme_args="$acme_args --debug"
+    fi
+    [ "$use_staging" -eq "1" ] && STATE_DIR="$STAGING_STATE_DIR" && staging="--staging"
+
+    set -- "$domains"
+    main_domain=$1
+
+    [ -n "$webroot" ] || [ -n "$dns" ] || pre_checks "$main_domain" || return 1
+
+    log "Running $APP for $main_domain"
+
+    if [ "$APP" = "uacme" ]; then
+	if [ ! -f  "$STATE_DIR/private/key.pem" ]; then
+	    log "Create a new ACME account with email $ACCOUNT_EMAIL use staging=$use_staging"
+	    $ACME $debug --confdir "$STATE_DIR" $staging --yes new $ACCOUNT_EMAIL
+	fi
+
+	local altdomain="$(for d in $domains; do echo -n " $d "; done)"
+
+	if [ -f "$STATE_DIR/$main_domain/cert.pem" ]; then
+	    log "Found previous cert config, use stating=$use_staging. Issuing renew."
+	    export CHALLENGE_PATH="$webroot"
+	    $ACME $debug --confdir "$STATE_DIR" $staging --never-create issue "$main_domain" $altdomain --hook=$HPROGRAM && ret=0 || ret=1
```

Comment:
```suggestion
	    $ACME $debug --confdir "$STATE_DIR" $staging --never-create issue $domains --hook=$HPROGRAM && ret=0 || ret=1
```
`$domains` has the full list we need here.